test/test_pg_client_store.js

```diff
@@ -0,0 +1,149 @@
+    db.hasName('test')
+      .then(
+        () => { should.ok(false); },
+        () => { done(); }
+      );
+  });
+
+  it('4. should add a test client', done => {
+    const client = { name: 'test', key: 'key1', secret: 'secret1' };
+    db.add(client).then(
+      (res) => {
+        res.name.should.be.equal('test');
+        res.key.should.be.equal('key1');
+        res.secret.should.be.equal('secret1');
+        done();
+      },
+      () => { should.ok(false); }
+    );
+  });
+
+  it('5. should have the `test` name', done => {
+    db.hasName('test')
+      .then(
+        (res) => {
+          res.name.should.be.equal('test');
+          res.key.should.be.equal('key1');
+          res.secret.should.be.equal('secret1');
+          done();
+        },
+        () => { should.ok(false); }
+      );
+  });
+
+  it('6. should have have one client registered', done => {
+    db.getAll()
+      .then(
+        (res) => {
+          res.length.should.be.equal(1);
+          res[0].name.should.be.equal('test');
+          res[0].key.should.be.equal('key1');
+          res[0].secret.should.be.equal('secret1');
+          done();
+        },
+        () => { should.ok(false); }
+      );
+  });
+
+  it('7. should add a second client', done => {
+    const client = { name: 'test2', key: 'key2', secret: 'secret2' };
+    db.add(client).then(
+      (res) => {
+        res.name.should.be.equal('test2');
+        res.key.should.be.equal('key2');
+        res.secret.should.be.equal('secret2');
+        done();
+      },
+      () => { should.ok(false); }
+    );
+  });
+
+  it('8. should have have two client registered', done => {
+    db.getAll()
+      .then(
+        (res) => {
+          res.length.should.be.equal(2);
+          res[0].name.should.be.equal('test');
+          res[0].key.should.be.equal('key1');
+          res[0].secret.should.be.equal('secret1');
+          res[1].name.should.be.equal('test2');
+          res[1].key.should.be.equal('key2');
+          res[1].secret.should.be.equal('secret2');
+          done();
+        },
+        () => { should.ok(false); }
+      );
+  });
+
+  it('9. should fail to add a client with the same name', done => {
+    const client = { name: 'test2', key: 'key2', secret: 'secret2' };
+    db.add(client).then(
+      (res) => {
+        should.ok(false);
+      },
+      () => { done(); }
+    );
+  });
+
+  it('10. should remove a client by key', done => {
+    db.removeByKey('key1').then(
+      done,
+      () => { should.ok(false); }
+    );
+  });
+
+  it('11. should have have one client registered', done => {
+    db.getAll()
+      .then(
+        (res) => {
+          res.length.should.be.equal(1);
+          res[0].name.should.be.equal('test2');
+          res[0].key.should.be.equal('key2');
+          res[0].secret.should.be.equal('secret2');
+          done();
+        },
+        () => { should.ok(false); }
+      );
+  });
+
+  it('12. should succeed to cleanup db', done => {
```

Comment:
Move this to a `after` block. You can check that the DB is actually cleared, but you are already testing that at the beginning of the suite, so it's ok if you remove test # 13.